### ISSUE TITLE
bookmarks deleted every startup, re-synced

### ISSUE THREAD

**Issue by bipedalist** [xbrowsersync/app#51]
### Steps to reproduce
1. start up browser
2. bookmarks sync starts by deleting all bookmarks, then resyncs and recreates
3. rinse, repeat

### Expected behaviour
Tell us what should happen
bookmarks are synced without being deleted

### Actual behaviour
Tell us what happens instead
all bookmarks are deleted as if xbrowsersync was doing the initial sync

### Client app information
**App Platform:** (Chrome, Firefox, iOS, etc) 
firefox v.63
**App Platform Version:** 
kubuntu
windows 10
windows 7
#### Other info

happening on every system where I am running xbrowser sync, three PC's
```
Enter any other information you think would be helpful here.
on this machine at least, getting repeated xbrowser sync error windows popping up



```

**Comment by nero120** [comment]
@bipedalist this is expected behaviour. When synced, xBrowserSync will check for updates on browser startup and if it finds that the local bookmarks are out of sync it will download the latest and refresh the local bookmarks. To do this it has to delete all of the local bookmarks first and then re-populate the local bookmarks based on the synced data.

In the future there may be scope to sync without deleting first but this is not the highest priority.

**Comment by bipedalist** [comment]
Okay, thanks. Here is what I am looking at this morning and wondering if this is normal behavior.

I've just started up a Win7 box (one of 3 PC's on this login account), and booted Firefox. The xbrowsersync icon on FF menu bar is greyed so appears inactive. 

When I open the xsync pulldown window, the 'welcome' pane appears with my sync ID already present, the password field is blank. I've previously installed, enabled and synced on this box so that seems anomalous. It seems like the extension is forgetting my previous run through the setup so it's doing this again. 

Maybe I am misunderstanding the behavior of the extension? Is it that I will be logging in, deleting all bookmarks and then repopulating them each sync?

**Comment by bipedalist** [comment]
I just checked the 3 PC's I am working on. 
1 Kubuntu pc: xbrowser sync is still logged in and active
1 Kubuntu box with virtual box/Windows 10; the Kubuntu FF xbsync is active and logged in, the Win10 extension is at the welcome window and not enabled
1 Windows 7 box - xbrowsersync is at welcome window, not enabled, fresh sync

All of these were previously enabled, synced and running so it seems like the Windows FF extensions are not staying logged in between reboot/restarts. 

I'm working on the Win7 box now. The xbsync extension has finished its fresh update of the bookmarks from being enabled. I probably should restart firefox and see if the extension goes disabled and then perhaps test a reboot as well.

**Comment by bipedalist** [comment]
Restarted Firefox and sync disables and back at welcome window.

![image](https://user-images.githubusercontent.com/44816142/48216759-6a8c7b80-e33a-11e8-8bc6-29300b8dea07.png)

Maybe I'm just going too fast. I don't understand why the Win boxes are doing this and the Kubuntu OS FF's are not.

**Comment by nero120** [comment]
Are you clearing browser cache in FF on close? If so it will also clear your xBrowserSync login data - this is a known issue (#38) which will be fixed in the next version (note this also affects Chrome). That would explain why FF is starting with sync disabled.

**Comment by bipedalist** [comment]
Windows is my main working environment, where I tend to use privacy and ad blocking extensions, like Duck Duck Go, which very well may be doing that. I will try turning them off between restarts. Thanks.

**Comment by nero120** [comment]
Ok @bipedalist, if that's the case once v.1.4.1 is released this should no longer be an issue! 😄

**Comment by nero120** [comment]
@bipedalist these issues should be resolved in the upcoming v1.5.0 release. If you want to try out this functionality I am running a [beta](https://github.com/xbrowsersync/app/releases/tag/v1.5.0beta) for a short time before the final release.

**Comment by nero120** [comment]
Please try again using the latest [beta](https://github.com/xbrowsersync/app/releases/tag/v1.5.0beta). I've included more logging and [raised a bug with Mozilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1556427) that affects moving or deleting multiple bookmarks simultaneously which could be a potential source of your issues.

If you do experience the issue again, please attach the xBrowserSync logs from all affected browsers.

**Comment by nero120** [comment]
@bipedalist would you mind trying the latest [v1.5.0beta11](https://github.com/xbrowsersync/app/releases/tag/v1.5.0beta)?

Please be aware however, due to a [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1556427) in Firefox's bookmarks API, changing multiple bookmarks simultaneously will likely result in an error in xBrowserSync. The issues has been fixed in the latest [Firefox Nightly (v69.0a1)](https://www.mozilla.org/en-US/firefox/nightly/all/) so please download and test with Nightly so that you don't hit that known issue.

**Comment by nero120** [comment]
@bipedalist I've just released [v1.5.0](https://github.com/xbrowsersync/app/releases/tag/v1.5.0) which resolves most syncing issues, however there are [known issues](https://github.com/xbrowsersync/app/wiki/Known-issues) in Firefox due to bug's in Firefox's bookmarks API which can cause changes to be missed when performing many changes or simultaneous changes.

Unfortunately there's not much we can do until Mozilla fixes these issues. I am tracking the relevant bugs (e.g. [1556427](https://bugzilla.mozilla.org/show_bug.cgi?id=1556427)) and will keep the pressure on Mozilla to resolve them asap.
